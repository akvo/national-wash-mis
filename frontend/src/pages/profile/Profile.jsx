import React, { useMemo } from "react";
import "./style.scss";
import { Space, Card, Divider, Row, Tag } from "antd";
import { store } from "../../lib";
import { Breadcrumbs, DescriptionPanel } from "../../components";
import { ProfileTour } from "./components";
import moment from "moment";

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      onchange: null,
      addListener: function () {},
      removeListener: function () {},
      addEventListener: function () {},
      removeEventListener: function () {},
      dispatchEvent: function () {},
    };
  };

const descriptionData =
  "This page shows your current user setup. It also shows the most important activities for your current user setup";

const Profile = () => {
  const { forms, user: authUser } = store.useState((s) => s);
  // const { trained } = authUser;

  const trainedBadge = useMemo(() => {
    if (authUser?.trained) {
      return (
        <Tag color="warning" style={{ marginBottom: 11 }}>
          Trained
        </Tag>
      );
    }
  }, [authUser?.trained]);

  const pagePath = [
    {
      title: "Control Center",
      link: "/control-center",
    },
    {
      title:
        (
          <Space align="center" size={15}>
            {authUser?.name}
            {trainedBadge}
          </Space>
        ) || "Profile",
    },
  ];

  const fullAdministrationName = window.dbadm
    .find((x) => x.id === 322)
    ?.full_name?.split("|")
    .join(" - ");

  return (
    <div id="profile">
      <Row justify="space-between">
        <Breadcrumbs pagePath={pagePath} />
        <ProfileTour />
      </Row>
      <DescriptionPanel description={descriptionData} />
      <Divider />
      <Card style={{ padding: 0, marginBottom: 12 }}>
        <h1>My Profile</h1>
        <ul className="profile-detail">
          <li data-testid="name">
            <h3>Name</h3>
            <Space size="large" align="center">
              <span>{authUser?.name}</span>
              <span style={{ fontStyle: "italic" }}>
                {authUser?.role?.value}
              </span>
            </Space>
          </li>
          <li>
            <h3>Phone Number</h3>
            <Space size="large" align="center">
              <span>{authUser?.phone_number}</span>
            </Space>
          </li>
          <li data-testid="role">
            <h3>Role</h3>
            <Space size="large" align="center">
              <span>{authUser?.role?.value}</span>
            </Space>
          </li>
          <li data-testid="organisation">
            <h3>Organization</h3>
            <Space size="large" align="center">
              <span>{authUser?.organisation?.name}</span>
            </Space>
          </li>
          <li data-testid="designation">
            <h3>Designation</h3>
            <Space size="large" align="center">
              <span>{authUser?.designation?.name}</span>
            </Space>
          </li>
          <li data-testid="administration">
            <h3>Administration</h3>
            <p>{fullAdministrationName || authUser?.administration?.name}</p>
          </li>
          <li data-testid="questionnaire">
            <h3>Questionnaires</h3>
            <Space size="large" align="center">
              {forms.map((qi, qiI) => (
                <span key={qiI}>{qi.name}</span>
              ))}
            </Space>
          </li>
          <li data-testid="last-login">
            <h3>Last login</h3>
            <Space size="large" align="center">
              <span>
                {authUser?.last_login
                  ? moment
                      .unix(authUser.last_login)
                      .format("MMMM Do YYYY, h:mm:ss a")
                  : "-"}
              </span>
            </Space>
          </li>
        </ul>
      </Card>
    </div>
  );
};

export default React.memo(Profile);
