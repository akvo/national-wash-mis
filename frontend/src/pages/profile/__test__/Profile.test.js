import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import Profile from "../Profile";

describe("pages/Profile", () => {
  test("test profile heading", () => {
    const { container, getByTestId } = render(<Profile />, {
      wrapper: MemoryRouter,
    });
    expect(
      screen.getByText(
        "This page shows your current user setup. It also shows the most important activities for your current user setup"
      )
    ).toBeInTheDocument();
    expect(screen.getByText("My Profile")).toBeInTheDocument();
    expect(getByTestId("name")).toBeInTheDocument();
    expect(getByTestId("role")).toBeInTheDocument();
    expect(getByTestId("organisation")).toBeInTheDocument();
    expect(getByTestId("designation")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
