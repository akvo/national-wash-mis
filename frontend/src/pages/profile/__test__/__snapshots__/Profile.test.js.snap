// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Profile test profile heading 1`] = `
<div>
  <div
    id="profile"
  >
    <div
      class="ant-row ant-row-space-between"
    >
      <div
        class="ant-breadcrumb"
      >
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            <a
              href="/control-center"
            >
              <h1
                class="ant-typography"
                style="display: inline;"
              >
                Control Center
              </h1>
            </a>
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            &gt;
          </span>
        </span>
        <span>
          <span
            class="ant-breadcrumb-link"
          >
            <h2
              class="ant-typography"
              style="display: inline;"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-center"
              />
            </h2>
          </span>
          <span
            class="ant-breadcrumb-separator"
          >
            &gt;
          </span>
        </span>
      </div>
      <button
        class="ant-btn ant-btn-text ant-btn-icon-only"
        type="button"
      >
        <span
          aria-label="question-circle"
          class="anticon anticon-question-circle"
          role="img"
          style="font-size: 24px; color: rgb(25, 144, 255);"
        >
          <svg
            aria-hidden="true"
            data-icon="question-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="description-paragraph"
    >
      This page shows your current user setup. It also shows the most important activities for your current user setup
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
    <div
      class="ant-card ant-card-bordered"
      style="padding: 0px; margin-bottom: 12px;"
    >
      <div
        class="ant-card-body"
      >
        <h1>
          My Profile
        </h1>
        <ul
          class="profile-detail"
        >
          <li
            data-testid="name"
          >
            <h3>
              Name
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 24px;"
              >
                <span />
              </div>
              <div
                class="ant-space-item"
              >
                <span
                  style="font-style: italic;"
                />
              </div>
            </div>
          </li>
          <li>
            <h3>
              Phone Number
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </li>
          <li
            data-testid="role"
          >
            <h3>
              Role
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </li>
          <li
            data-testid="organisation"
          >
            <h3>
              Organization
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </li>
          <li
            data-testid="designation"
          >
            <h3>
              Designation
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <span />
              </div>
            </div>
          </li>
          <li
            data-testid="administration"
          >
            <h3>
              Administration
            </h3>
            <p />
          </li>
          <li
            data-testid="questionnaire"
          >
            <h3>
              Questionnaires
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <span>
                  CLTS
                </span>
              </div>
            </div>
          </li>
          <li
            data-testid="last-login"
          >
            <h3>
              Last login
            </h3>
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
              >
                <span>
                  -
                </span>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
